lib/beamchmark/suite/measurements/cpu_info.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Beamchmark.Suite.Measurements.CpuInfo do
+  @moduledoc """
+  Module representing statistics about cpu usage.
+
+  Method of measuring:
+    - Take a snapshot of cpu usage every `@timeout` milliseconds
+    - Calculate the average cpu usage of processor (combining each core usage)
+    - At the end combine the results and calculate the average
+
+  **Warning!**
+    This module can give unstable cpu usage values when measuring a short time because of a high cpu volatility.
+    TODO Can be improved by taking average of 5-10 values for each snapshot
+  """
+
+  use Bunch.Access
+  alias Beamchmark.Math
+
+  @typedoc """
+  Single core snap statistics gathered via snapshot
+  """
+  @type cpu_core_usage_t ::
+          %{(core_id :: integer()) => usage :: Math.percent_t() | Math.percent_diff_t()}
+
+  @typedoc """
+  All information gathered via single snapshot + processor average
+  """
+  @type cpu_usage_t ::
+          %{
+            cpu_usage: cpu_usage :: [cpu_core_usage_t()],
+            average_all_cores: average_all_cores :: Math.percent_t()
+          }
+
+  @typedoc """
+  All information gathered via all snapshots
+  `all_average` is average from all snapshots
+  """
+  @type t :: %__MODULE__{
+          cpu_snapshots: [cpu_usage_t()],
+          average_all: Math.percent_t()
+        }
```

Comment:
I would remove cpu_core_usage_t and instead do something like 
```suggestion
  @typedoc """
  All information gathered via single snapshot + processor average
  """
  @type cpu_snapshot_t ::
          %{
            cpu_usage: %{(core_id :: integer()) => usage :: Math.percent_t() | Math.percent_diff_t() },
            avg: Math.percent_t()
          }

  @typedoc """
  All information gathered via all snapshots
  `all_average` is average from all snapshots
  """
  @type t :: %__MODULE__{
          cpu_snapshots: [cpu_snapshot_t()],
          avg: Math.percent_t()
        }
```